torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -119,6 +128,78 @@ def main_hook(self):
         self.zero.step()
 
 
+class _DDPBucket(NamedTuple):
```

Comment:
(can be done in followup PRs) This conceptually overlaps with DDP's Gradient Bucket. We probably want to let DDP expose its GradBuckets instead of creating new classes?

Reply:
I think that it _may_ be possible to get rid of this class altogether with how DDP's `GradBucket` is exposed currently. I can investigate this in a follow-up PR.